In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('../input/data11/advertising.csv')
df.head(20)

User ID  Gender   Age  EstimatedSalary  Purchased
0   15624510    Male  19.0          19000.0          0
1   15810944    Male  35.0          20000.0          0
2   15668575  Female  26.0          43000.0          0
3   15603246  Female  27.0          57000.0          0
4   15804002    Male  19.0          76000.0          0
5   15728773    Male  27.0          58000.0          0
6   15598044  Female  27.0          84000.0          0
7   15694829  Female  32.0         150000.0          1
8   15600575    Male  25.0          33000.0          0
9   15727311  Female  35.0          65000.0          0
10  15570769  Female  26.0          80000.0          0
11  15606274  Female  26.0          52000.0          0
12  15746139    Male  20.0          86000.0          0
13  15704987    Male  32.0          18000.0          0
14  15628972    Male  18.0          82000.0          0
15  15697686    Male  29.0          80000.0          0
16  15733883    Male  47.0          25000.0          1
17  15617482    Male  45.0          26000.0          1
18  15704583    Male  46.0          28000.0          1
19  15621083  Female  48.0          29000.0          1

In [3]:
X = df.iloc[:,2:4]
y = df['Purchased']

In [4]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size = 0.25,random_state = 5)

In [5]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train) 
X_test = scaler.transform(X_test)

In [6]:
from sklearn.svm import SVC
classifier = SVC(kernel = 'linear',random_state = 0)
classifier.fit(X_train,y_train)

SVC(kernel='linear', random_state=0)

In [7]:
y_pred = classifier.predict(X_test)
y_pred

array([1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0,
       1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1,
       0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0])

In [8]:
from sklearn.metrics import accuracy_score
acc = accuracy_score(y_test,y_pred)
acc                                           # ---------(1)

0.85

In [9]:
# OR
classifier.score(X_test,y_test)

0.85

In [10]:
from sklearn.model_selection import GridSearchCV
parameters = [{'kernel':['linear'],'C':[10,50,150,250,350,500,700,1000]},
              {'kernel':['rbf'],'gamma':[0.001,0.008,0.01,0.1,0.5,0.8,0.9],'C':[10,50,150,250,350,500,700,1000]}]
gscv = GridSearchCV(estimator = classifier,param_grid = parameters,scoring  = 'accuracy',cv = 10)
gscv.fit(X_train,y_train)

GridSearchCV(cv=10, estimator=SVC(kernel='linear', random_state=0),
             param_grid=[{'C': [10, 50, 150, 250, 350, 500, 700, 1000],
                          'kernel': ['linear']},
                         {'C': [10, 50, 150, 250, 350, 500, 700, 1000],
                          'gamma': [0.001, 0.008, 0.01, 0.1, 0.5, 0.8, 0.9],
                          'kernel': ['rbf']}],
             scoring='accuracy')

In [11]:
print(gscv.best_params_,'\n',gscv.best_score_)

{'C': 250, 'gamma': 0.1, 'kernel': 'rbf'} 
 0.9100000000000001


In [12]:
# As we find that the best kernel is rbf instead of linear with gamma value of 0.1, so we replace classifier variable by--

classifier = SVC(kernel = 'rbf',gamma = 0.1,random_state = 0)
classifier.fit(X_train,y_train)

SVC(gamma=0.1, random_state=0)

In [13]:
y_pred = classifier.predict(X_test)
y_pred

array([1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0,
       1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1])

In [14]:
acc = accuracy_score(y_test,y_pred)
acc                                      # --------(2)

0.92

**Conclusion -  From (1) & (2) we find that the accuracy of our model increased to 0.92 from 0.85**